packages/openapi3/src/cli/actions/convert/transforms/transform-component-schemas.ts

```diff
@@ -24,141 +18,138 @@ import { getScopeAndName } from "../utils/get-scope-and-name.js";
-    doc: schema.description,
-    schema,
-  };
+    types.push(tsEnum);
+  }
 
-  types.push(union);
-}
+  function populateModel(
+    types: TypeSpecDataTypes[],
+    rawName: string,
+    schema: OpenAPI3Schema
+  ): void {
+    const { name, scope } = getScopeAndName(rawName);
+    const extendsParent = getModelExtends(schema, scope);
+    const isParent = getModelIs(schema, scope);
+    types.push({
+      kind: "model",
+      name,
+      scope,
+      decorators: [...getDecoratorsForSchema(schema)],
+      doc: schema.description,
+      properties: getModelPropertiesFromObjectSchema(schema),
+      additionalProperties:
+        typeof schema.additionalProperties === "object" ? schema.additionalProperties : undefined,
+      extends: extendsParent,
+      is: isParent,
+      type: schema.type,
+    });
+  }
 
-function populateModel(types: TypeSpecDataTypes[], name: string, schema: OpenAPI3Schema): void {
-  const extendsParent = getModelExtends(schema);
-  const isParent = getModelIs(schema);
-  types.push({
-    kind: "model",
-    ...getScopeAndName(name),
-    decorators: [...getDecoratorsForSchema(schema)],
-    doc: schema.description,
-    properties: getModelPropertiesFromObjectSchema(schema),
-    additionalProperties:
-      typeof schema.additionalProperties === "object" ? schema.additionalProperties : undefined,
-    extends: extendsParent,
-    is: isParent,
-    type: schema.type,
-  });
-}
+  function populateUnion(types: TypeSpecDataTypes[], name: string, schema: OpenAPI3Schema): void {
+    const union: TypeSpecUnion = {
+      kind: "union",
+      ...getScopeAndName(name),
+      decorators: getDecoratorsForSchema(schema),
+      doc: schema.description,
+      schema,
+    };
 
-function getModelExtends(schema: OpenAPI3Schema): string | undefined {
-  switch (schema.type) {
-    case "boolean":
-      return "boolean";
-    case "integer":
-      return getIntegerType(schema);
-    case "number":
-      return getNumberType(schema);
-    case "string":
-      return getStringType(schema);
+    types.push(union);
   }
 
-  if (schema.type !== "object" || !schema.allOf) {
-    return;
+  function populateScalar(types: TypeSpecDataTypes[], name: string, schema: OpenAPI3Schema): void {
+    types.push({
+      kind: "scalar",
+      ...getScopeAndName(name),
+      decorators: getDecoratorsForSchema(schema),
+      doc: schema.description,
+      schema,
+    });
   }
 
-  if (schema.allOf.length !== 1) {
-    // TODO: Emit warning - can't extend more than 1 model
-    return;
-  }
+  function getModelExtends(schema: OpenAPI3Schema, callingScope: string[]): string | undefined {
+    if (schema.type !== "object" || !schema.allOf) {
+      return;
+    }
 
-  const parent = schema.allOf[0];
-  if (!parent || !("$ref" in parent)) {
-    // TODO: Error getting parent - must be a reference, not expression
-    return;
-  }
+    if (schema.allOf.length !== 1) {
+      // TODO: Emit warning - can't extend more than 1 model
```

Comment:
It would be nice to add a diagnostic here, this will not be obvious.  Also, do we have a plan for dealing with this (like making these model aliases and mixing them in to the base model)?

Note: this is just a comment, should not block check-in.

Reply:
This logic goes away in #4232 (PR was getting large so didn't combine them).

What that PR does is merge any inline-schemas into the model's properties and spreads any referenced schemas into the model. If one (and only one) referenced schema has a discriminator, the model will extend instead of spread that.